### ISSUE TITLE
Unexpected size of output of Isomap transform

### ISSUE THREAD

**Issue by VPetukhov** [wildart/ManifoldLearning.jl#14]
Another behavior I can't explain on data from [example.txt](https://github.com/wildart/ManifoldLearning.jl/files/2566048/example.txt).

Load it with 
```julia
import DelimitedFiles: readdlm
using ManifoldLearning

example = readdlm("example.txt");
```
But when I run transform on different subsets, output size doesn't always match to size of input:
- `size(transform(Isomap, example[:, 1:500], d=2).proj, 2)` returns 500
- `size(transform(Isomap, example[:, 1:1000], d=2).proj, 2)` returns 999 *(why not 1000?)*
- `size(transform(Isomap, example[:, 1:1141], d=2).proj, 2)` returns 1141 *(again as expected)*

**Comment by wildart** [comment]
That is interesting. The algorithm is deterministic, so I might blame MDS reduction step for returning incomplete set of eigenvectors.

But, I did not get it.  I used `pre-1.0` branch.
```julia
julia> E = transform(Isomap, example[:, 1:1000], d=2)
Isomap(outdim = 2, neighbors = 12)

julia> size(E.proj)
(2, 1000)
```

**Comment by mafla** [comment]
```julia
using ManifoldLearning, Distances, Statistics

d = 14
k = 12
X = randn(14, 200)
M = transform(Isomap, X; k = k, d = d)
Y = projection(M)
idx = ccomponent(M)

DX = pairwise(Euclidean(), X)
R2 = zeros(d)
for i = 1:d
    DY = pairwise(Euclidean(), Y[1:i, :])
    R2[i] = cor(DX[:], DY[:])^2
end
R2
```

Gives me very often a DimensionMismatch("inconsistent lengths") problem. Interestingly, when this happens the  ccomponent(M) is unordered. This is why I cannot use it to index X and estimate the explained variances... There must be some problem here. Any hint would be highly appreciated.

**Comment by wildart** [comment]
I suspect bug in selection of largest connected component. These issues should be resolved as soon as I push #15 which corrects CC computation.

**Comment by wildart** [comment]
Fixed by #15

**Comment by VPetukhov** [comment]
Thank you so much!
